### ISSUE TITLE
Better locations

### ISSUE THREAD

**Issue by haselwarter** [Andromedans/andromeda#281]
The locations reported in errors are often not very helpful. In `tests/error_apply_non_function.m31`, we have the following:

```
(* Test whether application errors have sensible locations *)
let f = Type
let x = Type
let y = Type

fail f x y
```

The error reported points to the definition of `f` instead of the last line, so I guess the comment should be a TODO :)

**Comment by andrejbauer** [comment]
Also equip `var_name` with locations.

**Comment by SkySkimmer** [comment]
Fixed in #305.
